src/Microsoft.OData.Core/Json/ODataJsonWriterUtils.cs

```diff
@@ -98,32 +99,52 @@ internal static void ODataValueToString(StringBuilder sb, ODataValue value)
             if (value == null || value is ODataNullValue)
             {
                 sb.Append("null");
+                return;
             }
 
-            ODataCollectionValue collectionValue = value as ODataCollectionValue;
-            if (collectionValue != null)
+            if (value is ODataCollectionValue collectionValue)
             {
                 ODataCollectionValueToString(sb, collectionValue);
+                return;
             }
 
-            ODataResourceValue resourceValue = value as ODataResourceValue;
-            if (resourceValue != null)
+            if (value is ODataResourceValue resourceValue)
             {
                 ODataResourceValueToString(sb, resourceValue);
+                return;
             }
 
-            ODataPrimitiveValue primitiveValue = value as ODataPrimitiveValue;
-            if (primitiveValue != null)
+            if (value is ODataPrimitiveValue primitiveValue)
             {
-                if (primitiveValue.FromODataValue() is string)
+                object valueAsObject = primitiveValue.FromODataValue();
+                string valueAsString;
+                if (ODataRawValueUtils.TryConvertPrimitiveToString(valueAsObject, out valueAsString))
                 {
-                    sb.Append(string.Concat("\"", JsonValueUtils.GetEscapedJsonString(value.FromODataValue()?.ToString()), "\""));
+                    if (valueAsObject is string)
+                    {
+                        valueAsString = JsonValueUtils.GetEscapedJsonString(valueAsString);
+                    }
+
+                    if (valueAsObject is string || valueAsObject is byte[] || valueAsObject is DateTimeOffset || valueAsObject is Guid || valueAsObject is TimeSpan | valueAsObject is Date || valueAsObject is TimeOfDay)
+                    {
+                        sb.Append('"').Append(valueAsString).Append('"');
+                    }

```

Comment:
```suggestion
                    if (valueAsObject is byte[] || valueAsObject is DateTimeOffset || valueAsObject is Guid || valueAsObject is TimeSpan | valueAsObject is Date || valueAsObject is TimeOfDay)
                    {
                        sb.Append('"').Append(valueAsString).Append('"');
                    }

```

Reply:
@ElizabethOkerio Why did you drop, `valueAsObject is string || ` from the `if` statement? Your suggestion is not logically equivalent to the existing statement. We want to enclose the following with doubel quotes: `string`, `byte[]` as string, `DateTimeOffset` as string, `Guid` as string, `TimeSpan` as string, `Microsoft.Edm.Date` as string, `Microsoft.Edm.TimeOfDay` as string

Reply:
@gathogojr I moved it to the check for string. Where you escape if `valueObject` is `string`